src/librustc_parse/parser/mod.rs

```diff
@@ -666,11 +666,14 @@ impl<'a> Parser<'a> {
                             // bail with a suggestion
                             // https://github.com/rust-lang/rust/issues/72373
                             if self.prev_token.is_ident() && &self.token.kind == &token::DotDot {
+                                let msg = &format!("if you meant to bind the contents of \
+                                    the rest of the array pattern into `{}`, use `@`",
+                                    pprust::token_to_string(&self.prev_token));
                                 expect_err
-                                    .span_suggestion_short(
+                                    .span_suggestion_verbose(
                                         self.token.span,
-                                        &format!("maybe missing `@` in binding pattern?"),
-                                        "@..".to_string(),
+                                        msg,
+                                        "@ ..".to_string(),
```

Comment:
```suggestion
                                        " @ ..".to_string(),
```

Reply:
This works for the source as authored in the IR:

```
  |
5 |         [h, ref ts ..] => foo(c, n - h) + foo(ts, n),
  |                    ^^ expected one of `,`, `@`, `]`, or `|`
  |
help: if you meant to bind the contents of the rest of the array pattern into `ts`, use `@`
  |
5 |         [h, ref ts @ ..] => foo(c, n - h) + foo(ts, n),
  |                    ^^^^

```

but if the error includes a space, then this is not the proper formatting.  There will be two spaces:

```
error: expected one of `,`, `@`, `]`, or `|`, found `..`
 --> src/main.rs:5:20
  |
5 |         [h, ref ts ..] => foo(c, n - h) + foo(ts, n),
  |                    ^^ expected one of `,`, `@`, `]`, or `|`
  |
help: if you meant to bind the contents of the rest of the array pattern into `ts`, use `@`
  |
5 |         [h, ref ts  @ ..] => foo(c, n - h) + foo(ts, n),
  |                     ^^^^

error: aborting due to previous error
```

Reply:
needed to convert to " @ " with the change suggested in https://github.com/rust-lang/rust/pull/72534#discussion_r429698568